server/src/infra/repositories/media.repository.ts

```diff
@@ -23,11 +25,25 @@ export class MediaRepository implements IMediaRepository {
   }
 
   async resize(input: string | Buffer, output: string, options: ResizeOptions): Promise<void> {
-    await sharp(input, { failOn: 'none' })
-      .resize(options.size, options.size, { fit: 'outside', withoutEnlargement: true })
-      .rotate()
-      .toFormat(options.format)
-      .toFile(output);
+    if (typeof input === 'string' && unsupportedSharpTypes.has(mimeTypes.lookup(input))) {
+      this.logger.log(`Sharp doesn't support ${mimeTypes.lookup(input)} type, using imagemagick instead`);
+      return new Promise((resolve, reject) => {
+        im.convert([input, '-resize', `${options.size}x${options.size}\>`, output], (err, stdout) => {
```

Comment:
Also I think we should add `-quality` and `-strip` options

Reply:
out of curiosity, what's the interest of potential enlargement of a converted photo (by using the `^` parameter)
In the existing `sharp` call there's a `withoutEnlargement: true` parameter that's equivalent to `\>`

IMO most of the time photos will only be shrunk, but I'd keep `>` just in case.


adding `-quality` and `-strip` is a good idea 👍 In fact I found that there's a `-thumbnail` flag that does stripping and resizing at the same time

Reply:
I added it

Reply:
> what's the interest of potential enlargement of a converted photo (by using the `^` parameter)

The goal is thumbnails quality. We specify `options.size=250` for webp (timeline) and `1440` for jpeg (asset viewer). For example, if we resize without a `^` flag while generating a webp thumbnail for the image with landscape orientation, then the resulting thumbnail height will be less (and much less in the case of panoramas) than desired (`250`) and timeline thumbnails may look quite distorted

In the `sharp` call we use `{ fit: 'outside'}` option for this

Reply:
Got it, I suggest we use both then: `^\>` to ensure that resizing is done with respect to the shortest dimension and also to prevent unnecessary upscaling if for some reason an original image is less than `options.size` WDYT?

Reply:
Can we combine both `^` and `>`? Does ImageMagick handle resize properly if we specify both options? I think we should test the following cases:
 - Photos (portrait/landscape orientation) with one dimension less than `options.size`
 - Photos (portrait/landscape orientation) with both dimensions greater than `options.size`

Reply:
My tests for `source.jpg (1500x210)`:
|#| Command line | Thumbnail dimensions |
|--------|--------|--------|
| 1 | `magick convert source.jpg -resize '250x250' thumbnail.jpg` | `250x35` |
| 2 | `magick convert source.jpg -resize '250x250>' thumbnail.jpg` | `250x35` |
| 3 | `magick convert source.jpg -resize '250x250^' thumbnail.jpg` | `1786x250` |
| 4 | `magick convert source.jpg -resize '250x250^>' thumbnail.jpg` | `1500x210` |

My tests for `source.jpg (1600x900)`:
|#| Command line | Thumbnail dimensions |
|--------|--------|--------|
| 1 | `magick convert source.jpg -resize '250x250' thumbnail.jpg` | `250x141` |
| 2 | `magick convert source.jpg -resize '250x250>' thumbnail.jpg` | `250x141` |
| 3 | `magick convert source.jpg -resize '250x250^' thumbnail.jpg` | `444x250` |
| 4 | `magick convert source.jpg -resize '250x250^>' thumbnail.jpg` | `443x250` |

Looks like a combination of both flags (the 4th test) works as desired: shrinks and not enlarges

Reply:
cool, I did the same, here are my results:

|                                                         | New size | Width | Height | Upscale? | Too small? |
|---------------------------------------------------------|----------|:-----:|:------:|----------|------------|
| Original                                                |          |  1000 |    400 |          |            |
| convert image.jpg -thumbnail 250x250^\> converted.jpg   |      250 |   625 |    250 |   FALSE  |    FALSE   |
| convert image.jpg -thumbnail 500x500^\> converted.jpg   |      500 |  1000 |    400 |   FALSE  |    FALSE   |
| convert image.jpg -thumbnail 2000x2000^\> converted.jpg |     2000 |  1000 |    400 |   FALSE  |    FALSE   |
| convert image.jpg -thumbnail 250x250\> converted.jpg    |      250 |   250 |    100 |   FALSE  |    TRUE    |
| convert image.jpg -thumbnail 500x500\> converted.jpg    |      500 |   500 |    200 |   FALSE  |    TRUE    |
| convert image.jpg -thumbnail 2000x2000\> converted.jpg  |     2000 |  1000 |    400 |   FALSE  |    FALSE   |
| convert image.jpg -thumbnail 250x250^ converted.jpg     |      250 |   625 |    250 |   FALSE  |    FALSE   |
| convert image.jpg -thumbnail 500x500^ converted.jpg     |      500 |  1250 |    500 |   TRUE   |    FALSE   |
| convert image.jpg -thumbnail 2000x2000^ converted.jpg   |     2000 |  5000 |   2000 |   TRUE   |    FALSE   |